jspsych-plugins/plugin-survey.js

```diff
@@ -0,0 +1,170 @@
+let IlsSurveyPlugin = (function (jspsych) {
+    "use strict";
+
+    /**
+      This plugin is intended to simplify the survey part of experiments.
+      It lets you define a survey with free-form HTML and exclusion criteria based on the response data.
+     */
+    class IlsSurvey {
+	static info = {
+	    name: "IlsSurveyPlugin",
+	    parameters: {
+		/**
+		  Definition of survey fields. For each field, a label and options may be provided, e.g.:
+		  handedness: {label: 'Handedness', options: {right: "Right", left: "Left"}}
+		  The labels have no effect on the survey form, but are used in the review screen.
+		 */
+		fields: {
+		    type: jspsych.ParameterType.OBJECT,
+		    default: {},
+		},
+		/**
+		  HTML for survey, will be internally wrapped in a <form> element
+		 */
+		html: {
+		    type: jspsych.ParameterType.HTML_STRING,
+		    default: '',
+		},
+		/**
+		  Used for defining exclusion criteria. Provided function will be called with response data
+		  as sole argument. Returning true means the participant is excluded.
+		 */
+		exclusion: {
+		    type: jspsych.ParameterType.FUNCTION,
+		    default: {},
+		},
+		/**
+		  Message for excluded participants
+		 */
+		exclusionPrompt: {
+		    type: jspsych.ParameterType.HTML_STRING,
+		    default: 'Unfortunately you cannot participate in this experiment because you do not meet our criteria for participants',
+		},
+		/**
+		  Whether a review screen should be shown
+		*/
+		review: {
+		    type: jspsych.ParameterType.BOOL,
+		    default: true
+		},
+		/**
+		  Prompt for review screen
+		 */
+		reviewPrompt: {
+		    type: jspsych.ParameterType.STRING,
```

Comment:
I might be nicer to make this a jspsych.ParameterType.HTML_STRING. Currently the You'll get a <h4></h4> with the value of this parameter as innerHTML while viewing the review. I thought at first the header doesn't really stand out as greater as the fields. But, if it is an HTML string, the user may specify there own styling/formatting to the header.